examples/image_processing/curvature_diffusion.cpp

```diff
@@ -0,0 +1,99 @@
+/*******************************************************
+ * Copyright (c) 2017, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <stdio.h>
+#include <arrayfire.h>
+#include <af/util.h>
+#include <cstdlib>
+
+using namespace af;
+
+static const unsigned ITERS = 64;
+
+array normalize(const array &p_in)
+{
+    float mx = max<float>(p_in);
+    float mn = min<float>(p_in);
+    return (p_in-mn)/(mx-mn);
+}
+
+array sobelFilter(const array &p_in)
+{
+    int w = 5;
+    if (p_in.dims(0) <  512) w = 3;
+    if (p_in.dims(0) > 2048) w = 7;
+
+    int h = 5;
+    if (p_in.dims(0) <  512) h = 3;
+    if (p_in.dims(0) > 2048) h = 7;
+
+    array ker = gaussianKernel(w, h);
+    array smooth = convolve(p_in, ker);
+
+    for (unsigned i=1; i<ITERS; ++i)
+        smooth = convolve(smooth, ker);
```

Comment:
can't you make this a larger separable kernel? 64 iterations is a lot. How long does it take to execute this?

Reply:
I don't remember the exact times, it was quite some time back. It doesn't take much time on CUDA and OpenCL time. On CPU, it would be definitely longer time, I have to build & run them once again to tell how much time it takes exactly. Building has been a big pain due to the amount of time it takes after rebasing from master. So, I am addressing all other feedback first and then going do the runs.